subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/ivy/IvyDynamicRevisionRemoteResolveIntegrationTest.groovy

```diff
@@ -108,6 +108,263 @@ dependencies {
+                                   status[module] = st
+                                }
+                            }
+                        }
+                        println status
+                    }
+                }
+                details.result.status = status[id.toString()]
+            }
+          }
+"""
+        when:
+        def projectA1 = ivyHttpRepo.module("group", "projectA", "1.1").publish()
+        def projectA2 = ivyHttpRepo.module("group", "projectA", "1.2").publish()
+        def projectB1 = ivyHttpRepo.module("group", "projectB", "1.1").publish()
+        def projectB2 = ivyHttpRepo.module("group", "projectB", "2.2").publish()
+        ivyHttpRepo.module("group", "projectA", "2.0").publish()
+
+
+        and:
+        def statusFile = temporaryFolder.createFile("versions.status")
+        statusFile << '''group:projectA:1.1;release
+group:projectA:1.2;release
+group:projectB:1.1;release
+group:projectB:2.2;integration
+'''
+        server.expectGet("/repo/status.txt", statusFile)
+        expectGetDynamicRevision(projectA2)
+        expectGetDynamicRevision(projectB1)
+
+        then: "custom metadata rule prevented parsing of ivy descriptor"
+        checkResolve "group:projectA:1.+": "group:projectA:1.2", "group:projectB:latest.release": "group:projectB:1.1"
+        outputContains 'Providing metadata for group:projectB:2.2'
+        outputContains 'Providing metadata for group:projectB:1.1'
+        !output.contains('Providing metadata for group:projectA:1.1')
+
+        and: "remote status file parsed only once"
+        outputContains 'Parsing status file call count: 1'
+        !output.contains('Parsing status file call count: 2')
+
+        when: "resolving the same dependencies"
+        server.expectHead("/repo/status.txt", statusFile)
+        server.expectHead("/repo/status.txt", statusFile)
+        checkResolve "group:projectA:1.+": "group:projectA:1.2", "group:projectB:latest.release": "group:projectB:1.1"
+
+        then: "should parse the result from cache"
+        output.contains('Parsing status file call count: 1')
+
+        when: "force refresh dependencies"
+        executer.withArgument("-PrefreshDynamicVersions")
+        statusFile.text = '''group:projectA:1.1;release
+group:projectA:1.2;release
+group:projectB:1.1;release
+group:projectB:2.2;release
+'''
+        server.expectHead("/repo/status.txt", statusFile)
+        server.expectGet("/repo/status.txt", statusFile)
+        expectListVersions(projectA2)
+        expectGetDynamicRevision(projectB2)
+
+        then: "shouldn't use the cached resource"
+        checkResolve "group:projectA:1.+": "group:projectA:1.2", "group:projectB:latest.release": "group:projectB:2.2"
+        outputContains 'Providing metadata for group:projectB:2.2'
+        !output.contains('Providing metadata for group:projectB:1.1')
+        !output.contains('Providing metadata for group:projectA:1.1')
+
```

Comment:
I'd like to see some test coverage for `--offline` and `--refresh-dependencies` too.